lib/source/pl/lib/std/math.cpp

```diff
@@ -136,6 +149,45 @@ namespace pl::lib::libstd::math {
             runtime.addFunction(nsStdMath, "atanh", FunctionParameterCount::exactly(1), [](Evaluator *, auto params) -> std::optional<Token::Literal> {
                 return std::atanh(params[0].toFloatingPoint());
             });
+
+            /* helper functions */
+            runtime.addFunction(nsStdMath, "accumulate", FunctionParameterCount::between(3,5), [](Evaluator *ctx, auto params) -> std::optional<Token::Literal> {
+                auto start = params[0].toUnsigned();
+                auto end = params[1].toUnsigned();
+                auto size = params[2].toUnsigned();
+                AccumulationOperation op = ADD;
+                if(params.size() > 3) {
+                    op = static_cast<AccumulationOperation>(params[3].toUnsigned());
+                }
+                types::Endian endian = 0;
+                if(params.size() > 4) {
+                    endian = static_cast<types::Endian>(params[4].toUnsigned());
+                }
+
+                u128 result = 0;
+                u128 endAddr = end / size;
+                using namespace hlp::bf;
+                auto reader = BufferedReader(ctx);
+                reader.seek(start);
+                reader.setEndAddress(end);
+                for(u128 addr = start; addr < endAddr; ++addr) {
+                    auto bytes = reader.read(addr, size);
+                    // copy to u128
+                    u128 value = 0;
+                    std::memcpy(&value, bytes.data(), size);
```

Comment:
Make sure to limit `size` to be 16 at max, otherwise you'll get a potential buffer overflow here